rules/python/lang/weak_hash_adler32.yml

```diff
@@ -0,0 +1,47 @@
+imports:
+  - python_shared_lang_import1
+  - python_shared_lang_datatype
+patterns:
+  - pattern: |
+      $<FUNCTION>($<OPTIONAL_DATA_TYPE>$<...>)
+    filters:
+      - variable: FUNCTION
+        detection: python_shared_lang_import1
+        scope: cursor
+        filters:
+          - variable: MODULE1
+            values: [zlib]
+          - variable: NAME
+            values: [adler32]
+      - either:
+          - variable: OPTIONAL_DATA_TYPE
+            detection: python_shared_lang_datatype
+            scope: result
+          - not:
+              variable: OPTIONAL_DATA_TYPE
+              detection: python_shared_lang_datatype
+              scope: result
```

Comment:
IIRC the optional datatype stuff was mostly for dynamic severity, so we could remove it now that severity is static and just catch on any usage of the weak algorithms. 

Having said that, probably something we should change across all languages for consistency. What do you think?